src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubConfiguration.java

```diff
@@ -55,17 +59,59 @@ public GitHubConfiguration() {
     }
 
     @NonNull
-    public List<Endpoint> getEndpoints() {
+    public synchronized List<Endpoint> getEndpoints() {
```

Comment:
Yes, I have got confused by usage of `endpoints` as both local variable and field names. Not a best practice from the code readability PoV, and it may cause errors if somebody refactors the code incorrectly